db/db_impl/db_impl.cc

```diff
@@ -1308,27 +1312,52 @@ Status DBImpl::UnlockWAL() {
   return Status::OK();
 }
 
-void DBImpl::MarkLogsSynced(uint64_t up_to, bool synced_dir,
-                            const Status& status) {
+Status DBImpl::MarkLogsSynced(uint64_t up_to, bool synced_dir) {
   mutex_.AssertHeld();
-  if (synced_dir && logfile_number_ == up_to && status.ok()) {
+  if (synced_dir && logfile_number_ == up_to) {
     log_dir_synced_ = true;
   }
+  VersionEdit synced_wals;
   for (auto it = logs_.begin(); it != logs_.end() && it->number <= up_to;) {
-    auto& log = *it;
-    assert(log.getting_synced);
-    if (status.ok() && logs_.size() > 1) {
-      logs_to_free_.push_back(log.ReleaseWriter());
+    auto& wal = *it;
+    assert(wal.getting_synced);
+    if (logs_.size() > 1) {
+      if (immutable_db_options_.track_and_verify_wals_in_manifest) {
+        synced_wals.AddWal(wal.number,
+                           WalMetadata(wal.writer->file()->GetFileSize()));
```

Comment:
If we do it outside of the `if` block, then assume that there is just one log now, then everytime `SyncWAL` syncs this log, a VersionEdit will be written to MANIFEST. We only want to write one VersionEdit for syncing each WAL.

Reply:
> If we do it outside of the `if` block, then assume that there is just one log now, then everytime `SyncWAL` syncs this log, a VersionEdit will be written to MANIFEST.

I thought that's what we wanted (otherwise, you will only track closed WALs)?

Could you clarify under which conditions we want to log this in the MANIFEST?

Reply:
We only track closed WALs. The reason behind this is, if we log the syncing event to MANIFEST for every syncing, then there might be too many edits, and since logging to MANIFEST needs another sync, it will negatively impact write throughput if synced writes are frequent.